### PR TITLE
Remove the deprecated api usage in hdfs

### PR DESCRIPTION
**Pull request by Shoothzj**
### Motivation
Remove the deprecated api usage in hdfs

### Modifications
Use try with resources instead of `closeQuitely`
Remove the Long constructor

### Documentation
  
- [ ] no-need-doc 
  
little internal optimization

### CODE REVIEW

**Comment by Shoothzj** [comment]
@codelipenghui @sijie @BewareMyPower PTAL

**Comment by Shoothzj** [comment]
@hangc0276 PTAL

**Review by codelipenghui — APPROVED** [review]

**Comment by Shoothzj** [comment]
@merlimat @eolivelli PTAL

File: pulsar-io/hdfs2/src/main/java/org/apache/pulsar/io/hdfs2/AbstractHdfsConnector.java [hunk 1]
```diff
@@ -138,12 +137,8 @@ protected void checkHdfsUriForTimeout(Configuration config) throws IOException {
         }
         InetSocketAddress namenode = NetUtils.createSocketAddr(address, port);
         SocketFactory socketFactory = NetUtils.getDefaultSocketFactory(config);
-        Socket socket = null;
-        try {
-            socket = socketFactory.createSocket();
+        try(Socket socket = socketFactory.createSocket()) {
```
Comment (BewareMyPower) [inline pulsar-io/hdfs2/src/main/java/org/apache/pulsar/io/hdfs2/AbstractHdfsConnector.java:140]:
```suggestion
        try (Socket socket = socketFactory.createSocket()) {
```

Add the space. I think currently the checkstyle plugin is not applied to pulsar-io module.

**Review by BewareMyPower — CHANGES_REQUESTED** [review]

**Comment by BewareMyPower** [comment]
LGTM, just left a comment for the code format.

**Comment by Shoothzj** [comment]
> LGTM, just left a comment for the code format.

Fixed, PTAL

**Review by BewareMyPower — APPROVED** [review]

**Comment by hangc0276** [comment]
@Shoothzj You can use `/pulsarbot run-failure-checks` to rerun failed tests instead of `close and reopen` the PR, thanks.

**Comment by Shoothzj** [comment]
@merlimat @eolivelli @hangc0276 PTAL

**Review by hangc0276 — APPROVED** [review]

**Pull request merged**